src/compiler/checker.ts

```diff
@@ -9028,8 +9123,19 @@ module ts {
             }
 
             checkSourceElement(node.body);
-            if (node.type && !isAccessor(node.kind) && !node.asteriskToken) {
-                checkIfNonVoidFunctionHasReturnExpressionsOrSingleThrowStatment(node, getTypeFromTypeNode(node.type));
+            if (node.type) {
+                if (!isAccessor(node.kind) && !node.asteriskToken) {
+                    checkIfNonVoidFunctionHasReturnExpressionsOrSingleThrowStatment(node, getTypeFromTypeNode(node.type));
+                }
+                let type = getTypeFromTypeNode(<TypeGuardTypeNode>node.type);
+                if (type.flags & TypeFlags.TypeGuard && ~(<TypeGuardType>type).parameterIndex) {
+                    checkTypeAssignableTo(
+                        (<TypeGuardType>type).type,
+                        getTypeAtLocation(node.parameters[(<TypeGuardType>type).parameterIndex]),
+                        (<TypeGuardTypeNode>node.type).type, 
+                        Diagnostics.Type_0_is_not_assignable_to_type_1

```

Comment:
You can leave off the message here. I believe the one you passed is the default message.